contracts/standard/arbitration/composed-arbitrable/example/ArbitrableTokenList.sol

```diff
@@ -0,0 +1,594 @@
+                return;
+            }
+        } else { // Appeal.
+            fundDisputeCache.cost = agreement.arbitrator.appealCost(agreement.disputeID, agreement.extraData);
+
+            fundDisputeCache.appealing = true;
+            (fundDisputeCache.appealPeriodStart, fundDisputeCache.appealPeriodEnd) = agreement.arbitrator.appealPeriod(agreement.disputeID);
+            fundDisputeCache.appealPeriodSupported = fundDisputeCache.appealPeriodStart != 0 && fundDisputeCache.appealPeriodEnd != 0;
+            if (fundDisputeCache.appealPeriodSupported) {
+                if (now < fundDisputeCache.appealPeriodStart + ((fundDisputeCache.appealPeriodEnd - fundDisputeCache.appealPeriodStart) / 2)) // In the first half of the appeal period.
+                    require(_side == 0, "It is the losing side's turn to fund the appeal.");
+                else // In the second half of the appeal period.
+                    require(
+                        _side == 1 && _paidFees.loserFullyFunded[_paidFees.loserFullyFunded.length - 1],
+                        "It is the winning side's turn to fund the appeal, only if the losing side already fully funded it."
+                    );
+            } else require(msg.value >= fundDisputeCache.cost, "Fees must be paid in full if the arbitrator does not support `appealPeriod`.");
+        }
+
+        // Compute required value.
+        if (!fundDisputeCache.appealing) { // First round.
+            fundDisputeCache.requiredValueForSide = fundDisputeCache.cost / 2;
+        } else { // Appeal.
+            if (!fundDisputeCache.appealPeriodSupported)
+                fundDisputeCache.requiredValueForSide = fundDisputeCache.cost;
+            else if (_side == 0) // Losing side.
+                fundDisputeCache.requiredValueForSide = fundDisputeCache.cost + (2 * _paidFees.stake[_paidFees.stake.length - 1]);
+            else { // Winning side.
+                fundDisputeCache.expectedValue = _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][0] - _paidFees.stake[_paidFees.stake.length - 1];
+                fundDisputeCache.requiredValueForSide = fundDisputeCache.cost > _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][0] + fundDisputeCache.expectedValue ? fundDisputeCache.cost - _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][0] : fundDisputeCache.expectedValue;
+            }
+        }
+
+        // Take contribution.
+        if (_paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][_side] >= fundDisputeCache.requiredValueForSide)
+            fundDisputeCache.stillRequiredValueForSide = 0;
+        else
+            fundDisputeCache.stillRequiredValueForSide = fundDisputeCache.requiredValueForSide - _paidFees.totalContributedPerSide[_paidFees.totalContributedPerSide.length - 1][_side];
+
+
+        if(item.balance == item.challengeReward) { // Party is attempting to start a dispute
```

Comment:
> What if I want to pay a portion of the reward?

The way the the contract is constructed, fees can be crowdsourced but the challengeReward must be paid in full by the party making a request or attempting to raise a dispute. In fact `item.balance` only manages challengeReward such that it can hold only 3 possible values: `0`, `challengeReward` or `challengeReward * 2`.

This raises 2 important points:
1. Should there be a built-in mechanism for crowdfunding deposits as there is for fees? I don't think this is a good idea for now.
2. The contract doesn't currently have a mechanism for a challenger to git his deposit back if the crowdfunding of fees fail. This has to be implemented.

Reply:
The contract now sends the deposit according to ruling. The party that will receive the deposit is the party that funds more/fully his side of the dispute/appeal fees.

If the dispute of appeal is fully funded by both sides, the deposit will go to the party that wins.